### ISSUE TITLE
Server crashes since last update (v0.1.8.14)

### ISSUE THREAD

**Issue by squeletal** [webmilio/WebmilioCommons-Releases#1]
Since updating to v0.1.8.14, exceptions are appearing when I'm launching my server. While I can join and play just fine, other people lose connection when they join.

Logs when launching the server:
```
[20:28:40] [1/WARN] [tML]: Silently Caught Exception: 
System.FormatException: Unrecognized Guid format.
   at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
   at System.Guid.Parse(String input)
   at WebmilioCommons.Players.WCPlayerOnJoinWorld.set_Guid(String value) in WebmilioCommons\Players\WCPlayerOnJoinWorld.cs:line 11
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
   at WebmilioCommons.Networking.Packets.NetworkPacket.Receive(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\Packets\NetworkPacket.cs:line 139
   at WebmilioCommons.Networking.NetworkPacketLoader.HandlePacket(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\NetworkPacketLoader.cs:line 51
   at WebmilioCommons.WebmilioCommonsMod.HandlePacket(BinaryReader reader, Int32 whoAmI) in WebmilioCommonsMod.cs:line 94
   at Terraria.ModLoader.ModNet.HandleModPacket(BinaryReader reader, Int32 whoAmI, Int32 length)
   at Terraria.MessageBuffer.GetData(Int32 start, Int32 length, Int32& messageType)
   at Terraria.NetMessage.CheckBytes(Int32 bufferIndex)
   at Terraria.Netplay.<>c.<ServerLoop>b__46_2()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
   at System.Threading.Tasks.ThreadPoolTaskScheduler.TryExecuteTaskInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.TaskScheduler.TryRunInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.Task.InternalRunSynchronously(TaskScheduler scheduler, Boolean waitForCompletion)
   at Terraria.Netplay.OnUpdate()
   at Terraria.Main.DoUpdate(GameTime gameTime)
   at Terraria.Main.Update(GameTime gameTime)
   at Terraria.Main.DedServ()
   at Terraria.Program.LaunchGame_()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.WindowsLaunch.Main(String[] args)

[20:28:40] [1/WARN] [tML]: Silently Caught Exception: 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
   at WebmilioCommons.Networking.Packets.NetworkPacket.Receive(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\Packets\NetworkPacket.cs:line 139
   at WebmilioCommons.Networking.NetworkPacketLoader.HandlePacket(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\NetworkPacketLoader.cs:line 51
   at WebmilioCommons.WebmilioCommonsMod.HandlePacket(BinaryReader reader, Int32 whoAmI) in WebmilioCommonsMod.cs:line 94
   at Terraria.ModLoader.ModNet.HandleModPacket(BinaryReader reader, Int32 whoAmI, Int32 length)
   at Terraria.MessageBuffer.GetData(Int32 start, Int32 length, Int32& messageType)
   at Terraria.NetMessage.CheckBytes(Int32 bufferIndex)
   at Terraria.Netplay.<>c.<ServerLoop>b__46_2()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
   at System.Threading.Tasks.ThreadPoolTaskScheduler.TryExecuteTaskInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.TaskScheduler.TryRunInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.Task.InternalRunSynchronously(TaskScheduler scheduler, Boolean waitForCompletion)
   at Terraria.Netplay.OnUpdate()
   at Terraria.Main.DoUpdate(GameTime gameTime)
   at Terraria.Main.Update(GameTime gameTime)
   at Terraria.Main.DedServ()
   at Terraria.Program.LaunchGame_()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.WindowsLaunch.Main(String[] args)

[20:28:40] [1/WARN] [Terraria]: Error on message Terraria.MessageBuffer
```

And when my friend joins:
```
[20:29:18] [1/WARN] [tML]: Silently Caught Exception: 
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at WebmilioCommons.Networking.Packets.ModPlayerNetworkPacket`1.PreReceive(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\Packets\ModPlayerNetworkPacket.cs:line 18
   at WebmilioCommons.Networking.Packets.NetworkPacket.Receive(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\Packets\NetworkPacket.cs:line 136
   at WebmilioCommons.Networking.NetworkPacketLoader.HandlePacket(BinaryReader reader, Int32 fromWho) in WebmilioCommons\Networking\NetworkPacketLoader.cs:line 51
   at WebmilioCommons.WebmilioCommonsMod.HandlePacket(BinaryReader reader, Int32 whoAmI) in WebmilioCommonsMod.cs:line 94
   at Terraria.ModLoader.ModNet.HandleModPacket(BinaryReader reader, Int32 whoAmI, Int32 length)
   at Terraria.MessageBuffer.GetData(Int32 start, Int32 length, Int32& messageType)
   at Terraria.NetMessage.CheckBytes(Int32 bufferIndex)
   at Terraria.Netplay.<>c.<ServerLoop>b__46_2()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
   at System.Threading.Tasks.ThreadPoolTaskScheduler.TryExecuteTaskInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.TaskScheduler.TryRunInline(Task task, Boolean taskWasPreviouslyQueued)
   at System.Threading.Tasks.Task.InternalRunSynchronously(TaskScheduler scheduler, Boolean waitForCompletion)
   at Terraria.Netplay.OnUpdate()
   at Terraria.Main.DoUpdate(GameTime gameTime)
   at Terraria.Main.Update(GameTime gameTime)
   at Terraria.Main.DedServ()
   at Terraria.Program.LaunchGame_()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.WindowsLaunch.Main(String[] args)
```

When reverting to the last version, the problem disappears.

**Comment by webmilio** [comment]
Hey man, sorry for the late reply.

I've had this bug reported a few times, but its not a widespread thing it seems. Can I have more information as to your setup ?
* List of mods
* What OS you're on

If you can give me that, I can try reproducing the bug!

**Comment by webmilio** [comment]
Issue fixed as of 0.1.9.1.
